nixos/modules/services/networking/thelounge.nix

```diff
@@ -0,0 +1,76 @@
+{ pkgs, lib, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thelounge;
+  dataDir = "/var/lib/thelounge";
+  configJsData = "module.exports = " + builtins.toJSON (
+    { private = cfg.private; port = cfg.port; } // cfg.extraConfig
+  );
+in {
+  options.services.thelounge = {
+    enable = mkEnableOption "The Lounge web IRC client";
+
+    private = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Make your The Lounge instance private. You will need to configure user
+        accounts by using the (<command>thelounge</command>) command or by adding
+        entries in <filename>${dataDir}/users</filename>. You might need to restart
+        The Lounge after making changes to the state directory.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 9000;
+      description = "TCP port to listen on for http connections.";
+    };
+
+    extraConfig = mkOption {
+      default = {};
+      type = types.attrs;
+      example = literalExample ''{
+        reverseProxy = true;
+        defaults = {
+          name = "Your Network";
+          host = "localhost";
+          port = 6697;
+        };
+      }'';
+      description = ''
+        The Lounge's <filename>config.js</filename> contents as attribute set (will be
+        converted to JSON to generate the configuration file).
+
+        The options defined here will be merged to the default configuration file.
+        Note: In case of duplicate configuration, options from <option>extraConfig</option> have priority.
+
+        Documentation: <link xlink:href="https://thelounge.chat/docs/server/configuration" />
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.users.thelounge = {
+      description = "thelounge service user";
+      group = "thelounge";
+      isSystemUser = true;
```

Comment:
Why does this need to be a system user? System users will have id's below 1000, but I see no reason why that would have to be the case.

Reply:
This is essentially a hint towards a service running as this user right?
I don't see any reason to assign specific UID ranges aside from a distinction between services / real users.

Is the size of the system user range a concern here?

Reply:
I haven't thought of it this way, but I've never seen a service use this. And the range is indeed pretty limited, from 400 to 499: https://github.com/NixOS/nixpkgs/blob/def7d562e01430ab48f4ab84a6363bfdbe6e584a/nixos/modules/programs/shadow.nix#L13-L14

So I think it's best to not do that unless the service needs a uid below 1000 for some reason

Reply:
It turns out most services on my machine are in the <400 uid range. I've dropped the system user requirement.

Reply:
> It turns out most services on my machine are in the <400 uid range

Ah yeah, so that's another issue, see https://github.com/NixOS/nixpkgs/issues/60732#issuecomment-488829636

Admittedly it is a bit odd to have services use uid's above 1000, which commonly are used by users, but that's the only way to get around the range restrictions for now until we have a better solution.

Reply:
Although actually, this 100 uid limit is only a problem if a user enables more than 100 `isSystemUser` services for a NixOS machine.